### PR TITLE
Workspace management endpoints

### PR DESCRIPTION
**Pull request by volodymyr-sokol**

### CODE REVIEW

**Review by marcusn90 — APPROVED** [review]

File: application/auth/models.py [hunk 1]
```diff
@@ -10,11 +10,16 @@ class User(Base, Persistent):
...
     created_time = db.Column(db.DateTime(), nullable=False)
     jti = db.Column(db.String, nullable=True)
+    email = db.Column(db.String(120), unique=True, nullable=False)
 
     @classmethod
     def find_by_username(cls, username):
         return cls.query().filter_by(username=username).first()
 
+    @classmethod
+    def find(cls, **kwargs):
```
Comment (nestorsokil) [inline application/auth/models.py]: This is cool, I think we can move this to Persistent parent class

File: application/workspace/v1_controller.py [hunk 2]
```diff
@@ -72,7 +70,40 @@ def register_ws_v1(payload: WorkspaceRegisterRequest):
...
+    user = User.find(email=payload.email)
+    if user is None:
+        if WorkspaceInvitation.find(email=payload.email, ws_id=ws_id) is None:
+            WorkspaceInvitation(email=payload.email, ws_id=ws_id, start_date=datetime.datetime.now(), status=WorkspaceInvitationStatus.PENDING).save()
+    elif WorkspaceUserModel.find(user_id=user.id, ws_id=ws_id) is None:
+        # start_date = payload.start_date if payload.start_date else datetime.date.now()
+        start_date = datetime.datetime.now()
+        WorkspaceUserModel(user_id=user.id, ws_id=ws_id, start_date=start_date).save()
+
+    return "", 201
```
Comment (nestorsokil) [inline application/workspace/v1_controller.py:93]: I think I created a dataclass HttpMessage somewhere for such cases

Comment (marcusn90) [reply application/workspace/v1_controller.py:93]: Not sure if we need to use HttpMessage with graphql

Comment (nestorsokil) [reply application/workspace/v1_controller.py:93]: This handler is not graphql, i think

Comment (nestorsokil) [reply application/workspace/v1_controller.py:93]: This handler is not graphql, i think

Comment (marcusn90) [reply application/workspace/v1_controller.py:93]: It's not, but will remove it once we have graphql mutation/resolver for that

**Pull request merged**